lib/secret-tasks.js

```diff
@@ -47,6 +49,48 @@ function getCredentialTableDetails(options){
 }
 
 /**
+ * @description Get the correct kmsKey for creating a Credstash store
+ * @param {string} options.DeploymentCenter.secret.kmsKeyAlias - Alias for a KMS Key
+ * @param {string} options.DeploymentCenter.secret.kmsKeyId - KMS Key Id
+ * @return {object} options
+ */
+function getKMSKeyId(options = {}){
+	// if we have no options, secrets are not enabled, just return
+	if (!options.DeploymentCenter.secret)
+		return options;
+
+	var secretConfig = options.DeploymentCenter.secret;
+
+	// if we have a kmsKeyId, use that and return
+	if (secretConfig.kmsKeyId) {
+		secretConfig.kmsKey = secretConfig.kmsKeyId;
+		return options;
+	}
+
+	// If we have the alias, go look up the id since this is what
+	// we need to create a credstash store
+	var params = {};
+	return Q.ninvoke(kms, 'listAliases', params)
```

Comment:
There's no way to do a query by alias?  You really have to iterate through them all?  That's terrible

Reply:
Looks like you can call `kms.describeKey()` with an alias.  http://docs.aws.amazon.com/kms/latest/APIReference/API_DescribeKey.html

Reply:
http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/KMS.html#describeKey-property

Reply:
Ahh yes I did not see that, I saw it took keyId and did not realize that could be the alias. I'll make that change